### PR TITLE
tests: arch: arm: move test arm_interrupt to new ztest API

### PR DESCRIPTION
**Pull request by shaoanxli**
Migrate the testsuite tests/arch/arm/arm_interrupt to the new ztest API.

Signed-off-by: Shaoan Li <shaoanx.li@intel.com>

### CODE REVIEW

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c [hunk 1]
```diff
@@ -458,7 +458,7 @@ void test_arm_user_interrupt(void)
 #pragma GCC push_options
 #pragma GCC optimize("O0")
 /* Avoid compiler optimizing null pointer de-referencing. */
-void test_arm_null_pointer_exception(void)
+ZTEST(arm_interrupt, test_arm_null_pointer_exception)
```
Comment (chen-png) [inline tests/arch/arm/arm_interrupt/src/arm_interrupt.c:461]: maybe use macro ZTEST_SKIP_IFDEF better.

Comment (shaoanxli) [reply tests/arch/arm/arm_interrupt/src/arm_interrupt.c:461]: Can you explain when do I need to use this ZTEST_SKIP_IFDEF ?

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c [hunk 2]
```diff
@@ -412,7 +412,7 @@ static inline void z_vrfy_test_arm_user_interrupt_syscall(void)
 }
 #include <syscalls/test_arm_user_interrupt_syscall_mrsh.c>
 
-void test_arm_user_interrupt(void)
+ZTEST_USER(arm_interrupt,test_arm_user_interrupt)
```
Comment (povergoing) [inline tests/arch/arm/arm_interrupt/src/arm_interrupt.c:415]: The space required after the `,`

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c [hunk 3]
```diff
@@ -448,7 +448,7 @@ void test_arm_user_interrupt(void)
 #endif
 }
 #else
-void test_arm_user_interrupt(void)
+ZTEST_USER(arm_interrupt,test_arm_user_interrupt)
```
Comment (povergoing) [inline tests/arch/arm/arm_interrupt/src/arm_interrupt.c:451]: ditto

**Review by povergoing — CHANGES_REQUESTED** [review]
Also, there are some issues in your commit-msg:
```
1: UC2 Signed-off-by: must have a full name
2: B4 Second line is not empty: "Migrate the testsuite tests/arch/arm/arm_interrupt to the new ztest API."
```
https://github.com/zephyrproject-rtos/zephyr/runs/7174539960?check_suite_focus=true

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c (hunk 1 above)
Comment (chen-png) [reply tests/arch/arm/arm_interrupt/src/arm_interrupt.c:461]: > maybe use macro ZTEST_SKIP_IFDEF better.

sorry, please ignore this comment. 
this is a condition that if one config is not enabled, then test should be skipped, maybe we need to add a new macro Z_TEST_SKIP_IFNDEF for this condition.

File: tests/arch/arm/arm_interrupt/src/arm_interrupt.c (hunk 2 above)
Comment (shaoanxli) [reply tests/arch/arm/arm_interrupt/src/arm_interrupt.c:415]: Thanks for reminding.

**Review by yperess — DISMISSED** [review]

**Review by povergoing — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Pull request merged**